### PR TITLE
HDDS-9418. Consolidate CertificateClient handleCase handling

### PR DESCRIPTION
**Pull request by ChenSammi**
## What changes were proposed in this pull request?

1. To Consolidate DefaultCertficateClient and CommonCertificateClient, and consolidate handleCase logic in all CertificateClient. 
2. RECOVER state is removed, and is replaced by GETCERT state, to reduce the chance that a missing certificate or public key requires admin intervene to fix. 

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-9418

## How was this patch tested?

tested by existing unit test with some modification.

### CODE REVIEW

**Comment by ChenSammi** [comment]
@fapifta , @Galsza , could you help to review this patch?

File: hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/security/x509/certificate/client/CommonCertificateClient.java
Comment (fapifta) [inline hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/security/x509/certificate/client/CommonCertificateClient.java:1]: As I see, this class is not used anymore, can we just delete it? Or do I miss a usage here?

**Review by fapifta — COMMENTED** [review]
Thank you for posting this improvement and simplification @ChenSammi, all in all looks really great, I have noted two minor things I believe it does worth to take one more look at.

**Comment by ChenSammi** [comment]
@fapifta , thank you for the review.  Can you take another look of the new commit?

**Review by fapifta — APPROVED** [review]
Thank you for addressing my concerns @ChenSammi, I think we are good to go with this one!

**Pull request merged**